pkg/api/types.go

```diff
@@ -109,6 +109,16 @@ type ClusterMetadata struct {
 type AddonProfile struct {
 	Enabled bool              `json:"enabled"`
 	Config  map[string]string `json:"config"`
+	// Identity contains information of the identity associated with this addon.
+	// This property will only appear in an MSI-enabled cluster.
+	Identity *UserAssignedIdentity `json:"identity,omitempty"`
```

Comment:
The add-on facility used in aks-engine is different from the add-on facility used in AKS. Hence I am not sure how this can be used in AKS. I was expecting the facility to be more generic and be VMSS/VMA template specific. What is the missing piece piece connecting this ?

Reply:
When generating ARM template, AKS RP will also pass `aksEngineAPI.AddonProfile` to AKS-Engine. After adding this in AKS-Engine, AKS RP will add the resourceID of created identity in `AddonProfile.Identity` so that AKS-Engine can associate the identity to every agent node.

Reply:
Was wondering won't the deployment of the template add the add-on enabled to add-on manager and then add-on manager pull the corresponding images when you deploy the template ? Wouldn't that interfere with the helm installation of add-on in case of AKS ?

Reply:
> wouldn't the deployment of the template add the enabled aks-engine add-ons to add-on manager and then pull the corresponding images when deployed?

I guess not? I haven't seen such code in AKS-Engine. The deployment of the template will do nothing related to addon. And seems in current AKS-Engine, it just ignores all `AddonProfile`. @jackfrancis @ritazh  could you please help confirm? If I understand this correctly, the management of all addons is fully controlled by AKS RP.

Reply:
Seems AKS's addon is recorded in `ContainerService.Properties.AddonProfiles`, AKS-Engine's addon is recorded in `ContainerService.Properties.OrchestratorProfile.KubernetesConfig.Addons`. Currently AKS-Engine will do nothing about addons in `ContainerService.Properties.AddonProfiles`.

Reply:
I have a similar question as @kkmsft. The changes you are making updates the aks-engine addons, which will be deployed by the add-on manager. How does this work with AKS addons? Can you pls verify this?

Reply:
The `AddonProfile` struct is actually not used by aks-engine. aks-engine uses the `KubernetesAddon` struct which is composed as a child property of `KubernetesConfig`.